central/networkgraph/service/service_impl.go

```diff
@@ -311,9 +314,67 @@ func (s *serviceImpl) getNetworkGraph(ctx context.Context, request *v1.NetworkGr
 			node.QueryMatch = true
 		}
 	}
+
+	if request.IncludePolicies {
+		err := s.enhanceWithNetworkPolicyIsolationInfo(ctx, graph)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return graph, nil
 }
 
+func (s *serviceImpl) enhanceWithNetworkPolicyIsolationInfo(ctx context.Context, graph *v1.NetworkGraph) error {
+	var deploymentIds []string
+	for _, node := range graph.GetNodes() {
+		if node.GetEntity().GetType() == storage.NetworkEntityInfo_DEPLOYMENT {
+			deploymentIds = append(deploymentIds, node.GetEntity().GetId())
+		}
+	}
+
+	deploymentObjects, err := s.deployments.GetDeployments(ctx, deploymentIds)
```

Comment:
Add a TODO and a ticket to make this cheaper by just selecting the required files once Postgres ships. This will make it much cheaper and we will be able to remove the cluster namespace context by select distinct(clusterid, namespace)

Reply:
Done